lib/certificate_chain_serializer.rb

```diff
@@ -0,0 +1,33 @@
+class CertificateChainSerializer
+  PATTERN = /-----BEGIN CERTIFICATE-----(?:.|\n)+?-----END CERTIFICATE-----/.freeze
+
+  def self.load(chain)
+    return [] unless chain
+    chain.scan(PATTERN).map do |cert|
+      OpenSSL::X509::Certificate.new(cert)
+    end
+  end
+
+  def self.dump(chain)
+    return nil if chain.blank?
+
+    unless chain.respond_to?(:map)
```

Comment:
apologies for the confusion. we don't really need these. `Gem::Package.new(body, gem_security_policy).spec` will catch the issues of spec.cert_chain not being an array or OpenSSL::X509::CertificateError. these changes are introducing code climate warning.

Reply:
Our end goal here is that an attacker can push gems that were built manually (bypassing gem cli verifications). We want to just make sure that we don't end up storing something unintentional and avoid server-side errors. 

The attacker won't be able to reach Version.create (or new) line because `pull_spec` will catch these and already raises correct errors. Unless you can think of bypassing `Gem::Package.new` verification, we would not have the issues I previously raised in this function.